demo/src/main/java/com/novoda/storagepathfinder/demo/LandingActivity.kt

```diff
@@ -0,0 +1,44 @@
+package com.novoda.storagepathfinder.demo
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import com.novoda.storagepathfinder.StoragePath
+import kotlinx.android.synthetic.main.activity_landing.*
+import java.util.*
+
+class LandingActivity : AppCompatActivity() {
+
+    private lateinit var assetCloner: AssetCloner
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_landing)
+
+        assetCloner = DemoDependenciesFactory.createAssetCloner(applicationContext)
+        val storageInspector = DemoDependenciesFactory.createStorageInspector(applicationContext)
+
+        val deviceStoragePaths = ArrayList<StoragePath>()
+        deviceStoragePaths.add(storageInspector.primaryStorageBasePath)
+        deviceStoragePaths.add(storageInspector.primaryStorageApplicationPath)
+        deviceStoragePaths.addAll(storageInspector.secondaryStorageBasePath)
+        deviceStoragePaths.addAll(storageInspector.secondaryStorageApplicationPath)
+
+        device_storage_roots.layoutManager = LinearLayoutManager(applicationContext)
+        device_storage_roots.adapter = StoragePathsAdapter(
+                layoutInflater,
+                deviceStoragePaths,
+                onAddFileClicked
+        )
+    }
+
+    private val onAddFileClicked: (StoragePath) -> Unit = { deviceStoragePath ->
```

Comment:
If we declare this as a function that takes a `StoragePath` instead of a property then we can pass it as a method reference.